src/sql/workbench/services/connection/browser/connectionWidget.ts

```diff
@@ -648,7 +648,18 @@ export class ConnectionWidget extends lifecycle.Disposable {
 			let options = selectedAccount.properties.tenants.map(tenant => tenant.displayName);
 			this._azureTenantDropdown.setOptions(options);
 			this._tableContainer.classList.remove(hideTenantsClassName);
-			this.onAzureTenantSelected(0);
+
+			// If we have a tenant ID available, select that instead of the first one
+			if (this._azureTenantId) {
+				let tenant = selectedAccount.properties.tenants.find(tenant => tenant.id === this._azureTenantId);
+				if (tenant) {
+					this.onAzureTenantSelected(options.indexOf(tenant.displayName));
```

Comment:
Shouldn't you select the first option if you can't find the tenant in the list either?

(I'd also suggest printing an error to the console since that doesn't seem like an expected scenario)